nemo/collections/tts/torch/g2ps.py

```diff
@@ -235,3 +237,162 @@ def __call__(self, text):
+
+        if apply_to_oov_word is None:
+            logging.warning(
+                "apply_to_oov_word=None, This means that some of words will remain unchanged "
+                "if they are not handled by any of the rules in self.parse_one_word(). "
+                "This may be intended if phonemes and chars are both valid inputs, otherwise, "
+                "you may see unexpected deletions in your input."
+            )
+
+        super().__init__(
+            phoneme_dict=self.phoneme_dict, word_tokenize_func=word_tokenize_func, apply_to_oov_word=apply_to_oov_word,
+        )
+
+        self.ignore_ambiguous_words = ignore_ambiguous_words
+        self.heteronyms = (
+            set(self._parse_file_by_lines(heteronyms))
+            if isinstance(heteronyms, str) or isinstance(heteronyms, pathlib.Path)
+            else heteronyms
+        )
+        self.phoneme_probability = phoneme_probability
+        self._rng = random.Random()
+
+    @staticmethod
+    def _parse_as_cmu_dict(phoneme_dict_path):
+        """Loads IPA CMUdict, and generates a set of all valid symbols."""
+        g2p_dict = defaultdict(list)
+        symbols = set()
+
+        _alt_re = re.compile(r'\([0-9]+\)')
+        with open(phoneme_dict_path, 'r') as file:
+            for line in file:
+                if len(line) and ('A' <= line[0] <= 'Z' or line[0] == "'"):
+                    parts = line.split('  ')
+                    word = re.sub(_alt_re, '', parts[0])
+                    word = word.lower()
+
+                    pronunciation = parts[1].strip()
+                    symbols.update(pronunciation)  # This will insert each char individually
+                    g2p_dict[word].append(list(pronunciation))
+
+        return g2p_dict, symbols
+
+    @staticmethod
+    def _parse_file_by_lines(p):
+        with open(p, 'r') as f:
+            return [l.rstrip() for l in f.readlines()]
+
+    def is_unique_in_phoneme_dict(self, word):
+        return len(self.phoneme_dict[word]) == 1
+
+    def parse_one_word(self, word: str):
+        """Returns parsed `word` and `status` (bool: False if word wasn't handled, True otherwise).
+        """
+        if self.phoneme_probability is not None and self._rng.random() > self.phoneme_probability:
+            return word, True
+
+        # Punctuation (assumes other chars have been stripped)
+        if re.search("[a-zA-Z]", word) is None:
+            return list(word), True
+
+        # Heteronym
+        if self.heteronyms and word in self.heteronyms:
+            return word, True
+
+        # `'s` suffix (with apostrophe) - not in phoneme dict
+        if (
+            len(word) > 2
+            and word.endswith("'s")
+            and (word not in self.phoneme_dict)
+            and (word[:-2] in self.phoneme_dict)
+            and (not self.ignore_ambiguous_words or self.is_unique_in_phoneme_dict(word[:-2]))
+        ):
+            return self.phoneme_dict[word[:-2]][0] + ["z"], True
```

Comment:
why `z` not `ˈɛs`?

Reply:
I'm seeing most `...'S` entries ending with "z" in the IPA dictionary, e.g. `AARON'S  ˈɛɹənz`, `ADMIRAL'S  ˈædmɝəɫz`, `AIRWAY'S  ˈɛɹˌweɪz`.

I've just found some special cases like words that end with "...T'S" (like `AIRPORT'S  ˈɛɹˌpɔɹts`) or "...S'S" (like `ALDUS'S  ˈɑɫdɪsɪz`), I can add those as well.